lib/read_only/dl_esm_inf/README.md

```diff
@@ -1,29 +1,120 @@
-The following dependencies must be available:
-- The GOcean infrastructure library ``dl_esm_inf``. A stable
-  version of this is included in PSyclone.
-- The ReadOnly and PSyData base classes, which are included in
-  PSyclone. These Jinja templates are processed to create
-  the read-only verification code for integer, 32- and 64-bit
-  reals, and 2-dimensional arrays.
+This library uses the [PSyData API](
+https://psyclone.readthedocs.io/en/stable/psy_data.html) to interface
+with the application. The following dependencies must be available:
+
+- The [GOcean](https://psyclone.readthedocs.io/en/latest/gocean1p0.html)
+  infrastructure library ``dl_esm_inf``. A stable version of this is included
+  in the PSyclone repository as a Git submodule (see ["Installation"](
+  https://psyclone-dev.readthedocs.io/en/stable/working_practises.html#dev-installation)
+  in the PSyclone [Developer Guide](
+  https://psyclone-dev.readthedocs.io/en/stable) for details on working with
+  submodules). However, it is not included in the PSyclone [installation](
+  ./../../README.md#installation) and has to be cloned separately.
+
+- The ``ReadOnly`` (``read_only_base.jinja``) and ``PSyData``
+  (``psy_data_base.jinja``) base classes, which are included in PSyclone
+  installation. These Jinja templates are processed to create
+  the read-only verification code for ``integer``, 32- and 64-bit ``real``
+  scalars, and 2-dimensional ``real`` and ``integer`` arrays. The generated
+  Fortran modules, ``read_only_base.f90`` and ``psy_data_base.f90``, are then
+  used by the supplied ``read_only.f90`` module to create the wrapper library.
 
 ## Compilation
-A makefile is provided for compilation. The environment variables
-``$F90`` and ``$F90FLAGS`` can be set to point to the Fortran compiler
-and flags to use. They default to ``gfortran`` and the empty string.
-The dl_esm_inf library is also required. It defaults to the version included
-in PSyclone (see `../../../external/dl_esm_inf/finite_difference`),
-but you can overwrite this by setting ``$INF_DIR`` to point to a different
-location. The makefile will compile the library if required.
-
-A full runnable example can be found in
-``examples/gocean/eg7``.
+
+The library is compiled with ``make`` using the provided ``Makefile``. The
+environment variables ``$F90`` and ``$F90FLAGS`` can be set to point to the
+[Fortran compiler](./../../README.md#compilation) and flags to use. They
+default to ``gfortran`` and the empty string.
+
+The location of the ``dl_esm_inf`` library is specified using the
+environment variable ``INF_DIR``. It defaults to the relative
+path to location of the version included in PSyclone repository
+(``<PSYCLONEHOME>/external/dl_esm_inf/finite_difference``). This is
+not available in the PSyclone [installation](./../../README.md#installation)
+so the exact path **must be specified** during the compilation process, e.g.
+
+```shell
+make INF_DIR=<path/to/dl_esm_inf/finite_difference>
```

Comment:
There is a difference between make variables and environment variables. This command sets a make variable, while the text refers to the environment variable (which is what we typically use everywhere). I think it would be better to use:
```
INF_DIR=<path/to/dl_esm_inf/finite_difference> make
```

Reply:
Ok, I've changed this here and in the other READMEs. Please note that the reason I put `make` in front because this is how it is done in the `psy_data.rst` documentation, see e.g. here: https://github.com/stfc/PSyclone/blob/master/doc/user_guide/psy_data.rst#read-only-verification-library-for-lfric. I'm happy to change it there so it is consistent.

Reply:
I've also thrown this in as part of #1209. I think in the past I always used makefile variables, but in PSyclone generally we use environment variables.

Reply:
Ah, now I understand! Thanks for the explanation 😄